docs/lang_spec/Cadence Language Specification.md

```diff
@@ -0,0 +1,5798 @@
+
+- **EBNF**:
+  - The formal EBNF for struct declarations in Cadence is as follows:
+    ```ebnf
+    structDeclaration
+      : access 'struct' identifier conformances? '{' membersAndNestedDeclarations '}'
+      ;
+    
+    membersAndNestedDeclarations
+      : (fieldDeclaration | functionDeclaration | structDeclaration)*
+      ;
+  ```
+    - **access**: The access control for the struct (e.g., pub for public, priv for private).
+    - **identifier**: The name of the struct.
+    - **conformances**: Optional interfaces the struct conforms to.
+    - **membersAndNestedDeclarations**: Defines the fields, functions, and nested structs within the struct.
+
+
+- **Struct Behavior**:
+  - **Copying Semantics**: When a struct is assigned to a new variable or passed to a function, a copy of the struct is made. Changes to the copy do not affect the original struct.
+    ```cadence
+    let user1 = User(id: 1, name: "Alice")
+    let user2 = user1  // user1 is copied to user2
+    
+    user2.updateName(newName: "Bob")
+    // user1.name is still "Alice", user2.name is now "Bob"
+    ```
+  - **Access Control**: The visibility of a struct and its members is controlled using access modifiers. For example, pub makes a struct or field accessible outside its defining contract, while access(contract) restricts visibility to the contract where it is defined.
+
+- **Restrictions**:
+  - **Initialization**: All fields must be initialized exactly once in the initializer. Fields cannot be initialized directly in the declaration.
+  - **No Inheritance**: Structs in Cadence do not support inheritance. Instead, Cadence encourages the use of composition to build complex types.
+
+By using structs, developers can efficiently model data that needs to be copied and manipulated independently, while also benefiting from Cadence’s strict type system and resource-oriented programming features.
+
+#### Resource Declarations
+
+In Cadence, a resource is a composite type designed to model unique, consumable assets that have strict ownership and usage constraints. Resources are linear types, meaning they must be used exactly once, and they cannot be copied, discarded, or implicitly shared. This behavior is particularly well-suited for managing valuable digital assets, such as tokens or other forms of ownership, where ensuring uniqueness and preventing accidental loss or duplication is critical.
+
+Resources are declared using the resource keyword, followed by an identifier (the name of the resource), optional conformances to interfaces, and a block containing fields and functions that define the resource’s behavior. Like structs, the access level of a resource and its members is controlled by access modifiers, which dictate the visibility and accessibility of the resource’s elements.
+
+- **Syntax**:
+  - The basic syntax for a resource declaration is as follows:
+    ```cadence
+    access(resource) resource ResourceName {
```

Comment:
`access(resource)` is not a valid access modifier. See the implementation for a valid list: https://github.com/onflow/cadence/blob/master/ast/access.go